api/api_keys/serializers.py

```diff
@@ -36,3 +43,21 @@ def validate_is_admin(self, is_admin: bool):
                 "RBAC is not installed, cannot create non-admin key"
             )
         return is_admin
+
+    def to_representation(self, instance):
+        if self.context["view"].action == "retrieve":
+            is_admin = instance.is_admin if hasattr(instance, "is_admin") else False
+            if is_admin is False and not settings.IS_RBAC_INSTALLED:
+                raise serializers.ValidationError(
+                    "RBAC is not installed, cannot create non-admin key"
+                )
```

Comment:
This can just be done in a `validate_is_admin` method on the serializer.

Reply:
Also, I'm confused why we're validating this on retrieve, shouldn't this only need to be validated on create?

Reply:
Good catch @matthewelwell, this code is unnecessary, deleted.

I also found a bug, it was already solved, I had to create a PR for the RBAC: https://github.com/Flagsmith/flagsmith-rbac/pull/19